linera-service/src/cli_wrappers/docker.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::util::CommandExt;
+use anyhow::{Context, Result};
+use pathdiff::diff_paths;
+use std::path::PathBuf;
+use tokio::process::Command;
+
+pub struct DockerImage {
+    pub name: String,
+}
+
+impl DockerImage {
+    pub fn new(name: String) -> Self {
+        Self { name }
+    }
+
+    pub async fn build(&self, bin_path: &PathBuf, github_root: &PathBuf) -> Result<()> {
```

Comment:
So with this choice of API, you could have a `DockerImage` that is not built yet